pkg/agent/baker.go

```diff
@@ -938,31 +938,46 @@ const sysctlTemplateString = `# This is a partial workaround to this upstream Ku
 net.ipv4.tcp_retries2=8
 net.core.message_burst=80
 net.core.message_cost=40
-{{- if .CustomLinuxOSConfig}}{{ if .CustomLinuxOSConfig.Sysctls}}{{ if .CustomLinuxOSConfig.Sysctls.NetCoreSomaxconn}}
```

Comment:
issue was 3 levels of ifs, where the inner layer is per sysctl, but the outer 2 are for all sysctl/linux config.

so if you set any custom sysctl/linux config but did NOT override those 5 values, you didn't get our defaults. 

this was a regression from January's refactor to avoid using reflection - https://github.com/Azure/AgentBaker/commit/92060d06a657f38e8750433f7cbd41309dae54c4

Reply:
the reflection avoided the nested ifs

Reply:
Did this happen because we would get zero values for some of these fields, meaning that it would never end up setting defaults (cause of all the ifs)?

Reply:
expanding from the bad commit:
```go
{{- if .CustomLinuxOSConfig}}
  {{ if .CustomLinuxOSConfig.Sysctls}}
    {{ if .CustomLinuxOSConfig.Sysctls.NetCoreSomaxconn}}
    net.core.somaxconn={{.CustomLinuxOSConfig.Sysctls.NetCoreSomaxconn}}
    {{end}}
    // missing else
  {{end}}
   // missing else
{{- else}}
net.core.somaxconn=16384
{{- end}}
```

make sense now?

Reply:
why did I do that? not sure honestly...maybe we can simplify further